src/Microsoft.TestPlatform.Utilities/CodeCoverageRunSettingsProcessor.cs

```diff
@@ -0,0 +1,484 @@
+        /// <see cref="XPathNavigator"/> style path. If unable to select the requested node it adds
+        /// default settings along the path.
+        /// </summary>
+        /// 
+        /// <param name="currentRootNode">
+        /// The root node from the current settings document for the extraction.
+        /// </param>
+        /// <param name="defaultRootNode">
+        /// The corresponding root node from the default settings document.
+        /// </param>
+        /// <param name="pathComponents">The path components.</param>
+        /// 
+        /// <returns>The requested node if successful, <see cref="null"/> otherwise.</returns>
+        private XmlNode SelectNodeOrAddDefaults(
+            XmlNode currentRootNode,
+            XmlNode defaultRootNode,
+            IEnumerable<string> pathComponents)
+        {
+            var currentNode = currentRootNode;
+            var partialPath = new StringBuilder();
+            
+            partialPath.Append(".");
+
+            foreach (var component in pathComponents)
+            {
+                var currentPathComponent = "/" + component;
+
+                // Append the current path component to the partial path.
+                partialPath.Append(currentPathComponent);
+
+                // Extract the node corresponding to the latest path component.
+                var tempNode = this.ExtractNode(currentNode, "." + currentPathComponent);
+
+                // If the current node extraction is unsuccessful then add the corresponding
+                // default settings node and bail out.
+                if (tempNode == null)
+                {
+                    var defaultNode = this.ExtractNode(
+                        defaultRootNode,
+                        partialPath.ToString());
+
+                    var importedChild = currentNode.OwnerDocument.ImportNode(defaultNode, true);
+                    currentNode.AppendChild(importedChild);
+
+                    return null;
+                }
+
+                // Node corresponding to the latest path component is the new root node for the
+                // next extraction.
+                currentNode = tempNode;
+            }
+
+            return currentNode;
+        }
+
+        /// <summary>
+        /// Checks if we should process the current exclusion node.
+        /// </summary>
+        /// 
+        /// <param name="node">The current exclusion node.</param>
+        /// 
+        /// <returns>
+        /// <see cref="true"/> if the node should be processed, <see cref="false"/> otherwise.
+        /// </returns>
+        private bool ShouldProcessCurrentExclusion(XmlNode node)
+        {
+            const string attributeName = "mergeDefaults";
+            const string attributeValue = "false";
+
+            foreach (XmlAttribute attribute in node.Attributes)
+            {
+                if (attribute.Name == attributeName && attribute.Value == attributeValue)
```

Comment:
I would make it case insensitive by:

attribute.Value == attributeValue -> !bool.Parse(attribute.Value)

Reply:
I think it's better to use bool.TryParse instead just to make sure we're protected against invalid input.

Reply:
Yes. You are right. I forgot about this :)